main.js

```diff
@@ -0,0 +1,319 @@
+}
+
+console.log(getReversedString('Hello!'));
+console.log(getReversedString('Arrays'));
+
+// 5
+
+function isPalindrome(text) {
+    const textWithoutSpecialCharacters = text.replace(/[^a-z]/gi, '').toLowerCase();
+    const reversedText = textWithoutSpecialCharacters.split('').reverse().join('');
+
+    return textWithoutSpecialCharacters === reversedText;
+}
+
+console.log(isPalindrome('Kayak'));
+console.log(isPalindrome('Hello!'));
+console.log(isPalindrome('Was it a cat I saw'));
+
+// 6
+
+function countLetters(string) {
+    const letterCount = {};
+    const splittedAndCleanedString = string.replace(/[^a-zA-Z]/g, '').toLowerCase().split('');
+
+    for (i = 0; i < splittedAndCleanedString.length; i++) {
+        const letter = splittedAndCleanedString[i];
+
+        if (letterCount[letter] > 0) {
+            letterCount[letter] = letterCount[letter] + 1;
+        } else {
+            letterCount[letter] = 1;
+        }
+    }
+    return letterCount;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+
+const peopleArray = [
+    {
+        name: 'Adam',
+        age: 20
+    },
+    {
+        name: 'Amanda',
+        age: 5
+    },
+    {
+        name: 'John',
+        age: 75
+    },
+    {
+        name: 'Dave',
+        age: 15
+    }
+]
+
+function getYoungestPerson(people) {
+   let youngestPerson = people[0].age;
+
+   for (i = 1; i < people.length; i++) {
+       if (people[i].age < youngestPerson) {
+           return youngestPerson = people[i].age;
+       }
+   }
+   return youngestPerson;
+}
+
+function getOldestPerson(people) {
+    let oldestPerson = people[0].age;
+
+    for (i = 1; i < people.length; i++) {
+        if (people[i].age > oldestPerson) {
+            return oldestPerson = people[i].age;
+        }
+    }
+    return oldestPerson;
+}
+
+function getAgeDifference(people) {
+    const oldestPersonAge = getOldestPerson(peopleArray);
+    const youngestPersonAge = getYoungestPerson(peopleArray);
+
+    return oldestPersonAge - youngestPersonAge;
+}
+
+console.log(getAgeDifference(peopleArray));
+
+// 8 - codewars
+
+function getPositiveNumbersSum(numbersArray) {
+    let sum = 0;
+
+    for (i = 0; i < numbersArray.length; i++) {
+        let currentNumber = numbersArray[i];
+
+        if (currentNumber > 0) {
+            sum = sum + currentNumber;
+        }
+    }
+    return sum;
+}
+
+console.log(getPositiveNumbersSum([1,-2,3,4,5]));
+
+// 9
+
+function getSumOfSquaredNumbers(numbersArray){
+    let sum = 0;
+
+    for (i = 0; i < numbersArray.length; i++) {
+        let currentnumber = numbersArray[i];
+        sum = sum + (currentnumber * currentnumber);
+    }
+    return sum;
+}
+
+console.log(getSumOfSquaredNumbers([1, 2, 3]));
+
+// 10
+
+function getNameInitials(name){
+    const nameConvertedToArray = name.toUpperCase().split(' ');
+    return `${nameConvertedToArray[0][0]}.${nameConvertedToArray[1][0]}`;
```

Comment:
This is extremely difficult to read. Create two variables - `firstName` and `lastName`.